lib/runner_test.go

```diff
@@ -645,3 +646,20 @@ func TestRunCommandFailure(t *testing.T) {
 		t.Fatal("Expected to get an error, got 'nil'")
 	}
 }
+
+type concurrentBuffer struct {
+	mutex sync.Mutex
```

Comment:
I would recommend using RWMutex since readers are not being blocked by each other. Having said that not too sure how much you care about lock contention in your use case in the test.

Reply:
I thought about it and in this case there is no difference at all, as reads and writes won't happen concurrently. Writes, perhaps.